### ISSUE TITLE
[ENHANCEMENT] The logger strategy in Marqo 

### ISSUE THREAD

**Issue by wanliAlex** [marqo-ai/marqo#359]
**Is your feature request related to a problem? Please describe.**
we currently cannot modify the logger level in Marqo in an efficient way.

**Describe the solution you'd like**
we can change the logger level in Marqo. We should also have a reference table on the logger level and content.

design doc will be available soon.

**Comment by wanliAlex** [comment]
# Marqo Logger Design

##### TLDR
- 1. We move all the logs inside `add_documents` and `search` to debug level. The default level is still INFO but has a very clean log. (logs from `uvicorn` mostly).
- 2. If you want to read those logs, use `-e MARQO_LOG_LEVEL='debug'`.
- 3. If you want a even cleaner log, use `-e MARQO_LOG_LEVEL='warning'`. (Note: you don't even see `uvicorn` logs now).

## Overview
Currently, we have 3 loggers functions from Marqo (strictly speaking, we have more than 3 loggers as each module has its own logger. However, all the loggers are loaded and set from these 3 functions.) We also have 1 logger from `uvicorn`.

- 1. [tensor_search_logger](https://github.com/marqo-ai/marqo/blob/814e0fa162d309d2895f25c17b637cf4dd807d4c/src/marqo/tensor_search/tensor_search_logging.py#L4-L25) (to be discussed in this design)
- 2. [s2_inference_logger](https://github.com/marqo-ai/marqo/blob/814e0fa162d309d2895f25c17b637cf4dd807d4c/src/marqo/s2_inference/logger.py#L1-L12)
- 3. [Marqo logger](https://github.com/marqo-ai/marqo/blob/mainline/src/marqo/marqo_logging.py)
- 4. [Uvicorn logger](https://www.uvicorn.org/settings/#logging)

We can change the logger level inside these functions so all the loaded loggers will have the same level. That is to say, we can have 4 different environment variables to control the logger levels (which is not needed).

Here is a brief conclusion on the content of all the loggers.
#### Tensor_search_logger
This logger handles most of the information in Marqo, including `on_start_script.py`, `connections.py`,`index_meta_cache.py`, `tensor_search.py`,`redis_throttle.py`.
The timing logger is a logger loaded in `tensor_search.py`

#### Uvicorn_logger
This logger is controlled by `uvicorn`. It contains the http response. We should keep it at level INFO.
```python
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
INFO:     172.17.0.1:44838 - "POST /indexes/coco-2014-train/search?&device=cuda HTTP/1.1" 200 OK
```


#### S2_inference_logger
This logger logs the information including model loading, model utils. I don't think we need to modify the logger level here. Examples are:
```python
logger.info(f'loaded {model_name} on device {device} with normalization={normalize_embeddings}')
logger.info("Detecting custom clip model path. We use generic clip model loading.")
logger.info(f"currently only a single attribute can be reranked over for {model_name}. taking the first field {[searchable_attributes[0]]} from {searchable_attributes}")
```

#### Marqo_logger
This is the most light weight logger in Marqo. It is only used in `parallel.py`.  I don't think a variable is needed for this logger.
```python
logger.info(f'starting add documents using {self.n_chunks} chunks per process...')
logger.info(f"restricting threads to {self.threads_per_process} for process={self.process_id}")
logger.info(f'estimated time left for process {self.process_id} is {time_left} seconds ')
logger.info(f'took {end - start} sec for {self.n_docs} documents')
logger.info(f"found {n_documents} documents")
logger.info(f"using {n_processes} processes")
logger.info(f'Performing parallel now across devices {device_ids}...')
logger.info(f"finished indexing all documents. took {end - start} seconds to index {n_documents} documents")
```

## Considerations
For this `tensor_search_logger` level design, we should consider two things.

- 1. What is the default level of the logger? **logger level** 
- 2. What level should we use to log a specific information? **information level**

#### Examples
Currently, all the timing information is logged at INFO level or DEBUG level. That is to say we can easily filter them out by setting the logger level to WARNING. However, we will also **LOSS** the information from other modules such as:
```python
# In on_start_script.py
logger.info(f"found devices {device_names}")
logger.info("downloading nltk data....")
logger.info("completed loading nltk")
# In connections.py
logger.info(f"Took {((t1-t0)*1000):.3f}ms to connect to redis and load scripts.")
```
It is hard to claim that timing logs are as important as logs above. Therefore, we aim to move most of the logs to debug level.

**Comment by wanliAlex** [comment]
solved
